maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelCacheManager.java

```diff
@@ -0,0 +1,74 @@
+package org.apache.maven.model.building;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.xml.filter.DependencyKey;
+
+/**
+ * 
+ * @author Robert Scholte
+ * @since 3.7.0
+ */
+@Named
+@Singleton
+public class DefaultModelCacheManager implements ModelCacheManager
+{
+    private static final Map<Path, Model> MODELCACHE = Collections.synchronizedMap( new HashMap<Path, Model>() );
```

Comment:
Singleton so not static maybe? This kind of thing leaks easily otherwise when embed.

Also some invalidation should be added? LastModifiedDate?

Reply:
I agree on the static versus singleton, but people can still create it with a constructor. As long as the caches lives for 1 Maven run, I don't see any reason for invalidation. Also, disappearing models will cause issues later on, otherwise you need to re-read them again, which is more resource consuming

Reply:
The reason i had in mind was a plain vi on a file

Reply:
ModelCacheManager has been replaced with existing ReactorModelPool